chalk-parse/src/parser.lalrpop

```diff
@@ -72,9 +72,26 @@ TraitDefn: TraitDefn = {
 };
 
 AssocTyDefn: AssocTyDefn = {
-    "type" <name:Id> <p:Angle<ParameterKind>> ";" => AssocTyDefn {
-        name: name,
-        parameter_kinds: p
+    "type" <name:Id> <p:Angle<ParameterKind>> <b:(":" <Id> <Angle<Parameter>>)?>
+        <w:QuantifiedWhereClauses> ";" =>
+    {
```

Comment:
Ok so currently we do not handle at all the `+` separator which expresses multiple bounds as in `where T: Clone + Sized`. This is not a problem for e.g. where clauses on traits because you can still write `where T: Clone, T: Sized` but for associated types since there is no other way to express these multiple bounds, we need to be able to parse them with the `+` acting as a separator.

Reply:
done